libraries/provider_firewall_windows.rb

```diff
@@ -68,11 +68,29 @@ def whyrun_supported?
       windows_file.content build_rule_file(new_resource.rules['windows'])
       windows_file.run_action(:create)
 
-      # if the file was changed, restart iptables
-      if windows_file.updated_by_last_action?
+      # if the file was changed
+      # OR
+      # if current_rule_count is not equal to the number of rules in file
```

Comment:
I generally don't think we should try to check if the rules have been tampered with, and especially not by counting rules. Everywhere else in this cookbook, we assume someone isn't tampering with the rules outside of chef; I think the Windows functionality should also keep that assumption, or we get into counting rules, parsing rules, etc (all very error prone things to do).